adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -123,6 +125,34 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
   val sequences: SequenceDictionary
 
   /**
+   * Replaces the sequence dictionary attached to a GenomicRDD.
+   *
+   * @param newSequences The new sequence dictionary to attach.
+   * @return Returns a new GenomicRDD with the sequences replaced.
+   */
+  def replaceSequences(newSequences: SequenceDictionary): U
+
+  /**
+   * Appends sequence metadata to the current RDD.
+   *
+   * @param sequencesToAdd The new sequences to append.
+   * @return Returns a new GenomicRDD with the sequences appended.
+   */
+  def addSequences(sequencesToAdd: SequenceDictionary): U = {
+    replaceSequences(sequences ++ sequencesToAdd)
+  }
+
+  /**
+   * Appends metadata for a single sequence to the current RDD.
+   *
+   * @param sequenceToAdd The sequence to add.
+   * @return Returns a new GenomicRDD with this sequence appended.
+   */
+  def addSequence(sequenceToAdd: SequenceRecord): U = {
```

Comment:
My preference here is that if you have multiple records to add, you either provide them as a `SequenceDictionary` or chain multiple `addSequence` calls.